apps/history/declarativehistorymodel.cpp

```diff
@@ -100,6 +104,8 @@ QVariant DeclarativeHistoryModel::data(const QModelIndex & index, int role) cons
         return url.title().isEmpty() ? url.url() : url.title();
```

Comment:
Similarly, sorry for the odd placement.

Would there be scope to also remove an individual icon when the history link is removed? It adds complexity, because the icons are shared between all the history items for the same domain, so some check would be needed to ensure there are no others still in the list. If you go down this route it might also make sense to make `FaviconManager::sanitiseHostname()` accessible for the comparison.

Reply:
done, as sanitizedHostname()